### PR TITLE
Fix 2 issues that prevent auto_bind_ipv6 from happening on my setup

### PR DESCRIPTION
**Pull request by theraphim**
#### Type Of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:
- [x] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description

Without the first commit, on my setup - which has lots of interfaces with lots of addresses - auto_bind_ipv6 will make kamailio stuck on startup. It actually happens in get_flags, where data sent back from kernel can't fit into 8K anymore.

Second commit is fixing obvious typo while we're at it.

### CODE REVIEW

**Comment by sergey-safarov** [comment]
Is this fix this issue too?
https://github.com/kamailio/kamailio/issues/2404#issuecomment-661060298

when used
```
bind_ipv6_link_local=0
```

**Comment by theraphim** [comment]
> Is this fix this issue too?
> [#2404 (comment)](https://github.com/kamailio/kamailio/issues/2404#issuecomment-661060298)
> 
> when used
> 
> ```
> bind_ipv6_link_local=0
> ```

I am not sure. The whole identification code is too convoluted for a casual analysis. If you can apply and test, we'll get the answer back.

As I said, this whole part will benefit from a rewrite. It'll be smaller, easier to analyze, more efficient and less error prone - for example, right now it's performing almost the same exchange with netlink twice and it can't handle dropped netlink messages.

File: src/core/socket_info.c [hunk 1]
```diff
@@ -1184,6 +1189,16 @@ static int build_iface_list(void)
 		nll = 0;
 		p = buf;
 		while(1) {
+			// This (and the loop above in get_flags) is
+			// broken in more than one aspect. For example,
+			// if a receive is shorter than nlmsghdr,
+			// nlp->nlmsg_type will point at garbage.
+			// Proper fix for this (and static buffer sizing
+			// problem) would be to use libnl.
```
Comment (miconda) [inline src/core/socket_info.c:1197]: This comment does not belong in the source code.

If there is still any issue left after this commit, then report it in the issue tracker. The developers do not scan the code for comments pointing at possible issues.

Comment (theraphim) [reply src/core/socket_info.c]: Removed.

**Comment by miconda** [comment]
Thanks! I am going to merge it.

Related to the comment, at a quick look, `libnl` seems to be linux specific, not usable on macos or *bsds. Anyhow, as I said, if something still needs to be fixed/tuned, report on issue tracker.

**Pull request merged**